static/js/containers/CourseDrawer.js

```diff
@@ -0,0 +1,116 @@
+// @flow
+import React from "react"
+import R from "ramda"
+import { connect } from "react-redux"
+import { withRouter } from "react-router"
+import { Drawer, DrawerContent } from "rmwc/Drawer"
+import { Theme } from "rmwc/Theme"
+
+import { setShowCourseDrawer } from "../actions/ui"
+import { getViewportWidth } from "../lib/util"
+
+import type { Dispatch } from "redux"
+import type { Course } from "../flow/discussionTypes"
+import ExpandedCourseDisplay from "../components/ExpandedCourseDisplay"
+import { actions } from "../actions"
+
+type Props = {
+  showCourseDrawer: boolean,
+  dispatch: Dispatch<*>,
+  course: ?Course,
+  courseId: number
+}
+
+const shouldLoadData = R.complement(
+  R.allPass([
+    // if course id's don't match
+    R.eqProps("courseId"),
+    // courses don't match
+    R.eqProps("course")
+  ])
+)
+
+export class CourseDrawer extends React.Component<Props> {
+  width: number
+
+  constructor(props: Props) {
+    super(props)
+    this.width = getViewportWidth()
+  }
+
+  componentDidMount() {
+    const { courseId } = this.props
+    window.addEventListener("resize", () => this.onResize())
```

Comment:
can this just be `window.addEventListener("resize", this.onResize)`?

Reply:
Doesn't work, throws an error.  Currently the same as in the normal `Drawer` container.

Reply:
huh! well this works then I suppose.